packages/airnode-protocol/contracts/utils/SelfServeRrpBeaconServerWhitelister.sol

```diff
@@ -0,0 +1,123 @@
+    /// @param expirationTimestamps Array of expiration timestamps that correspond to the beaconIds
+    /// @param beaconIdsIndefiniteWhitelistStatus Array of beaconIds that have an associated indefinite whitelist status
+    /// @param indefiniteWhitelistStatuses Array of indefinite whitelist status that correspond to the beaconIds
+    constructor(
+        address rrpBeaconServerAddress,
+        bytes32[] memory beaconIdsExpirationTimestamp,
+        uint64[] memory expirationTimestamps,
+        bytes32[] memory beaconIdsIndefiniteWhitelistStatus,
+        bool[] memory indefiniteWhitelistStatuses
+    ) {
+        require(
+            rrpBeaconServerAddress != address(0),
+            "RrpBeaconServer address zero"
+        );
+        rrpBeaconServer = rrpBeaconServerAddress;
+        for (
+            uint256 ind = 0;
+            ind < beaconIdsExpirationTimestamp.length;
+            ind++
+        ) {
+            beaconIdToExpirationTimestamp[
+                beaconIdsExpirationTimestamp[ind]
+            ] = expirationTimestamps[ind];
+        }
+        for (
+            uint256 ind = 0;
+            ind < beaconIdsIndefiniteWhitelistStatus.length;
+            ind++
+        ) {
+            beaconIdToIndefiniteWhitelistStatus[
+                beaconIdsIndefiniteWhitelistStatus[ind]
+            ] = indefiniteWhitelistStatuses[ind];
+        }
+    }
+
+    /// @notice Adds a new beaconId with an expiration timestamp
+    /// that can be whitelisted by readers
+    /// @param beaconId The beaconId to set an expiration timestamp for
+    /// @param expirationTimestamp The expiration timestamp for the beaconId
+    function setBeaconIdWithExpirationTimestamp(
+        bytes32 beaconId,
+        uint64 expirationTimestamp
+    ) external override onlyOwner {
+        beaconIdToExpirationTimestamp[beaconId] = expirationTimestamp;
+        emit SetBeaconIdWithExpirationTimestamp(beaconId, expirationTimestamp);
+    }
+
+    /// @notice Adds a new beaconId with an indefinite whitelist status
+    /// that can be whitelisted by readers indefinetly
+    /// @param beaconId The beaconId to set an indefinite whitelist status for
+    /// @param indefiniteWhitelistStatus The indefinite whitelist status for the beaconId
+    function setBeaconIdWithIndefiniteWhitelistStatus(
+        bytes32 beaconId,
+        bool indefiniteWhitelistStatus
+    ) external override onlyOwner {
+        beaconIdToIndefiniteWhitelistStatus[
+            beaconId
+        ] = indefiniteWhitelistStatus;
+        emit SetBeaconIdWithIndefiniteWhitelistStatus(
+            beaconId,
+            indefiniteWhitelistStatus
+        );
+    }
+
+    /// @notice Whitelists a reader on the RrpBeaconServer with an expiration timestamp
+    /// @param beaconId The beaconId to whitelist
+    /// @param reader The reader to whitelist on the beaconId
+    function whitelistReaderWithExpiration(bytes32 beaconId, address reader)
```

Comment:
I think it's simpler to have a single `whitelistReader()` function that both sets the whitelist expiration and the indefinite whitelist status

Reply:
okay makes sense